### DISCUSSION TITLE
When does `mutateAsync` have a new reference on every re-render ?

### DISCUSSION THREAD

**Discussion by 0xmkh** [TanStack/query#7812] [Q&A]
Hey there! I have a quick question about the `useMutation` hook. 

Earlier today i wanted to add `mutateAsync` in the dependency array of `useEffect`, but wasn't sure if it'd have a new reference on every re-render. I would love to know at what point does `mutateAsync` become a new function reference when a re-render occurs.

**Comment by TkDodo** [comment]
`mutateAsync` does not become a new reference every render, but the object returned from `useMutation` is new
